pkg/agent/informant.go

```diff
@@ -199,16 +199,26 @@ func NewInformantServer(
 
 		// we need to spawn these in separate threads so the caller doesn't block while holding
 		// runner.lock
-		runner.spawnBackgroundWorker(context.TODO(), shutdownName, func(c context.Context) {
-			if err := httpServer.Shutdown(c); err != nil {
+		runner.spawnBackgroundWorker(ctx, shutdownName, func(context.Context) {
+			// we want shutdown to (potentially) live longer than the request which
+			// made it, but having a timeout is still good.
+			ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
```

Comment:
20 seconds is a _super_ long timeout, compared to other things here (or: compared to our usual configuration of them). I'd either: make this shorter (e.g. 5s), add a config field for it, or calculate it from an existing config field

Reply:
I mean I think the question is more "how long could an HTTP request handler take to do a thing in normal operation and double(ish) it.

The cap on this is about 30s in my mind (which is probably just what the timeout handler was in System V init scripts between sigterm and sigkill if the process doesn't die, and which has definitely been carried further into the future.